src/validation.cpp

```diff
@@ -1046,9 +1046,6 @@ static bool AcceptToMemoryPoolWorker(CTxMemPool& pool, CValidationState& state,
         if(tpcin <= 0)
             return state.Invalid(false, REJECT_INSUFFICIENTFEE, "bad-txns-token-without-fee");
 
-        if(tpcin - tpcout <= 0)
-            return state.Invalid(false, REJECT_INSUFFICIENTFEE, "bad-txns-token-insufficient");
-
```

Comment:
Isn't this check necessary?

Reply:
This is done in CheckTxInputs (and cause 'bad-txns-in-belowout' error). when I keep this check some test cases were failing because of the difference in error code.